### PR TITLE
Allow --init to identify a wider range of URLs in `repository.url`

### PR DESCRIPTION
**Pull request by charliewxyz**
### Rationale

Previously, `git://` scheme was not identified in `repository.url` when running `--init`

### Changes

- Allow `--init` to identify URL in `repository.url` with:
  - any scheme (`https://` or `git://`)
  - any domain name (e.g. with `www.` or without)
  - either suffix `.git` or without

### Version change

**Patch version update** (`2.0.1`): just bug fixes, no breaking changes, no new features

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -60,14 +60,14 @@ function getRepositoryUrl(repository, version) {
 	if (typeof repository === 'object' && repository.url) {
 		// "https://github.com/user/repo.git"
 		[, type, name] = repository.url.match(
-			/https?:\/\/(.+)(?:\.com|\.net|\.org)\/(.+\/.+)\.git/
+			/^(?:\w+?:\/\/)?(.+)(?:\.\w+(?:\.\w+)*)\/([\w-]+\/[\w-]+)(?:\.git)?$/
```
Comment (charliewxyz) [inline lib/index.js]: Not sure if I should add comment:

```js
/**
 * Group 1. (?:\w+?:\/\/)? - any protocol ending with `://`, e.g. `git://`, optional, not captured
 * Group 2. (.+) - lower level domains, e.g. `www.github`, required, captured
 * Group 3. (?:\.\w+(?:\.\w+)*) - top level domains, e.g. `.com` or `.co.uk`, required, not captured
 * Group 4. ([\w-]+\/[\w-]+) - username and repository name ([a-zA-Z0-9_-]), e.g. `nikolajevp/changelog-updater`, required, captured
 * Group 5. (?:\.git)? - suffix `.git`, not required, not captured
 */
```

File: lib/index.js [hunk 2]
```diff
@@ -60,14 +60,14 @@ function getRepositoryUrl(repository, version) {
...
 		);
 	}
 
-	switch (type) {
-		case 'github':
+	switch (true) {
+		case type.includes('github'):
```
Comment (karniol) [inline lib/index.js:68]: Because we might have lower level domain such as `www.github`.

File: lib/index.js (hunk 1 above)
Comment (charliewxyz) [reply lib/index.js]: Simplified regex. Added comment.

**Review by nikolajevp — APPROVED** [review]

**Comment by nikolajevp** [comment]
Thanks!

**Pull request merged**